Sources/StreamVideo/Controllers/CallController.swift

```diff
@@ -75,137 +122,119 @@ class CallController: @unchecked Sendable {
         ring: Bool = false,
         notify: Bool = false
     ) async throws -> JoinCallResponse {
-        let response = try await joinCall(
-            create: create,
-            callType: callType,
-            callId: callId,
-            options: options,
-            migratingFrom: migratingFrom,
-            ring: ring,
-            notify: notify
-        )
-
-        currentSFU = response.credentials.server.edgeName
-        statsInterval = TimeInterval(response.statsOptions.reportingIntervalMs / 1000)
-        let settings = callSettings ?? response.call.settings.toCallSettings
-        
-        try await connectToEdge(
-            response,
-            sessionID: sessionID,
-            callType: callType,
-            callId: callId,
-            callSettings: settings,
-            ring: ring,
-            migratingFrom: migratingFrom
+        try await webRTCCoordinator.connect(
+            callSettings: callSettings,
+            ring: ring
         )
-        
-        setupStatsTimer()
-        
+        guard
+            let response = try await joinCallResponseSubject
+            .nextValue(dropFirst: 1, timeout: 15)
```

Comment:
if we have several cases like this, it might make sense to extract the timeout in a constant.

Reply:
We will as I mentioned earlier.